redbot/core/bot.py

```diff
@@ -134,9 +140,165 @@ def __init__(self, *args, cli_flags=None, bot_dir: Path = Path.cwd(), **kwargs):
 
         self._permissions_hooks: List[commands.CheckPredicate] = []
 
+    @property
+    def cog_mgr(self) -> NoReturn:
+        raise AttributeError("Please don't mess with the cog manager internals.")
+
+    @property
+    def uptime(self) -> datetime:
+        """ Allow access to the value, but we don't want cog creators setting it """
+        return self._uptime
+
+    @uptime.setter
+    def uptime(self, value) -> NoReturn:
+        raise RuntimeError(
+            "Hey, we're cool with sharing info about the uptime, but dont try and assign to it please."
+        )
+
+    @property
+    def db(self) -> NoReturn:
+        raise AttributeError(
+            "We really don't want you touching the bot config directly. "
+            "If you need something in here, take a look at the exposed methods "
+            "and use the one which corresponds to your needs or "
+            "open an issue if you need an additional method for your use case."
+        )
+
+    @property
+    def counter(self) -> NoReturn:
+        raise AttributeError(
+            "Please make your own counter object by importing ``Counter`` from ``collections``."
+        )
+
+    @property
+    def color(self) -> NoReturn:
+        raise AttributeError("Please fetch the embed color with `get_embed_color`")
```

Comment:
Maybe specify `bot.get_embed_color` for this and `colour`.

Reply:
Not entirely sure this would be better. We can't be sure they are using a variable named `bot` here. Leaving this one as-is as it is mostly for developers, who should be able to find where it happens from the stack trace.